pkg/virt-handler/heartbeat/ksm.go

```diff
@@ -291,7 +318,12 @@ func handleKSM(node *v1.Node, clusterConfig *virtconfig.ClusterConfig) (ksmLabel
 	return true, ksm.running
 }
 
-func disableKSM(node *v1.Node) {
+func disableKSM(getNode func() (*v1.Node, error)) {
+	node, err := getNode()
+	if err != nil {
+		return
+	}
```

Comment:
I'd prefer to leave it as it was, leaving the `getNode` handling to the caller

Reply:
Are you sure? This will make things a lot more verbose:
```go
	if ksmConfig == nil {
		if enabled {
			node, err := getNode()
			if err != nil {
				return false, false
			}
			disableKSM(node)
		}
```

But I'm open to this approach if you think it's better.

Reply:
This shows that we do change the logic. Previously if `Get` for `Node` would fail we would short circuit the hearbeat, now we could fail, not disable KSM and report it is disabled?

Reply:
Thank you @xpivarc, you have a valid point there.

> now we could fail, not disable KSM and report it is disabled?

We could also fail with the previous code, but you're right that at the time you were writing the comment if the `Get()` call would fail then KSM would still be reported as disabled for 10 minutes.

I've now added a retry mechanism in the [last commit](https://github.com/kubevirt/kubevirt/pull/14632/commits/2f2fd5accab6e8fe1fd629d531a7a60bdc047cce), it basically runs `handleKSM()` every minute as long as it fails for any reason. However, if it succeeds, the next run would be only if kubevirt's CR config change or after 20 minutes.

WDYT?